src/Symfony/Component/Config/Tests/Resource/ReflectionClassResourceTest.php

```diff
@@ -174,6 +193,20 @@ public static function getSubscribedEvents()
     }
 }
 
+class TestMessageSubscriber implements MessageSubscriberInterface
+{
+    public static function getHandledMessages(): iterable
+    {
+        foreach (TestMessageSubscriberConfigHolder::$handledMessages as $key => $subscribedMessage) {
+            yield $key => $subscribedMessage;
+        }
+    }
+}
+class TestMessageSubscriberConfigHolder
+{
+    public static $handledMessages = [];
+}
```

Comment:
I discovered that `testEventSubscriber()` and `testServiceSubscriber()` are invalid. Part of the signature of `ReflectionClassResource` is the constants and their values. So, to purely test that the resource becomes stale when a static method is called, we must do that *without* modifying any constants.

Those other 2 methods need to be fixed in 3.4 - you can comment out the code in `ReflectionClassResource` for each and they still pass.